spokestack/wakeword/ring_buffer.py

```diff
@@ -110,16 +109,21 @@ def read(self) -> np.ndarray:
         if self.is_empty:
             raise IndexError("Buffer is empty")
 
-        # pull item from current tail index
         item = self._buffer[self._read : self._read + 1]
-        # calc new tail index
         self._read = (self._read + 1) % self._max_length
         return item
 
-    def to_array(self) -> np.ndarray:
+    def read_all(self) -> np.ndarray:
         """ Dumps the entire contents of the buffer to an array
 
         Returns: Array with full contents of the buffer
 
         """
-        return self._buffer[: self.capacity]
+
+        self.rewind()
+        full = np.empty_like(self._buffer[: self.capacity])
```

Comment:
this assumes that the buffer is full when you read it
if the buffer isn't full, you'll return a numpy array with the whole capacity, and the client would have to filter it for empties to determine the number of items read
a simpler way to do this would be to read the buffer elements into a list, then call `np.stack` to concatenate them
also, the buffer class has a dtype parameter, so it would be better to convert the results to that instead of `np.float32`